docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -993,6 +994,74 @@ The `-fexceptions` option is also needed for C code that needs to interoperate w
 
 ---
 
+### Enable run-time checks for C++ virtual function pointers corruption
+
+| Compiler Flag                                               | Supported since  | Description                                                                                                    |
+|:------------------------------------------------------------|:----------------:|:---------------------------------------------------------------------------------------------------------------|
+| <span id="-fvtable-verify">`-fvtable-verify=std`</span>     | GCC 4.9.4        | Enable run-time checks for C++ virtual function pointers corruption after shared libraries have been loaded  |
+| <span id="-fvtable-verify">`-fvtable-verify=preinit`</span> | GCC 4.9.4        | Enable run-time checks for C++ virtual function pointers corruption before shared libraries have been loaded |
+| <span id="-fvtable-verify">`-fvtable-verify=none`</span>    | GCC 4.9.4        | Disable enable run-time checks for C++ virtual function pointers corruption.                                 |
+
+#### Synopsis
+
+In object-oriented programming, object methods rely on dynamic dispatching to select the right virtual function to call at runtime. The vtable is the table of virtual function pointers linked to the code of the virtual function to be executed. The vtable can be overwritten by vtable hijacking attack. This option is used to check, at runtime, that each virtual function call is valid for its object type. This check prevents virtual function pointers from being corrupted or overwritten, and stops the application (SIGABRT) if the pointer integrity check fails. This virtual table pointer check is based on the data structure built at application start-up.
+
+This option has three choices[^gccvtv]:
+
+- `std` - Built after shared libraries have been loaded and initialized, vtable data imply to process check at normal object initialization time.
+- `preinit` - Built before shared libraries have been loaded and initialized, vtable data enable check before shared libraries are loaded to process check at early execution time.
+- `none` - turns off the vtable checks.
+
+[^gccvtv]: Team GCC, [Program Instrumentation Options: `-fvtable-verify`](https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#index-fvtable-verify), GCC Manual, August 2013
+
+### Performance implications
+
+Caroline Tice played an important role in integrating the vtable verification (VTV) by submitting the commit of this feature in GCC 4.9[^gccvtvcommit]. She involved on articles that give information about performance penalty. She is describe performance penalty following 3 points:
```

Comment:
There was feedback in the [C/C++ Compiler BP Guide Call in 2024-05-16](https://docs.google.com/document/d/1JY8FREBPCUUFpuv7-4B9EjeS2MLDpel0dbG5DFWrTns/edit#bookmark=id.cobg2blj7ukq) that the "_performance implications section reads very differently than text elsewhere in the guide, should be reworded to be more consistent with other sections"_

This is perhaps partially because, unlikely the rest of the guide, which aims to acknowledge the original author's through appropriate references. This section seems to feature the author rather prominently. Maybe this paragraph could be shortened while still making sure the author's original work is included as a reference.